target/linux/realtek/files-5.10/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -853,24 +853,17 @@ void rtnc_83xx_update_cntr(int r, int released)
 void rtnc_930x_update_cntr(int r, int released)
 {
 	int pos = (r % 3) * 10;
-	u32 reg = RTL930X_DMA_IF_RX_RING_CNTR + ((r / 3) << 2);
-	u32 v = sw_r32(reg);
+	u32 reg = rtnc_930x_dma_if_rx_ring_cntr(r);
 
-	v = (v >> pos) & 0x3ff;
-	pr_debug("RX: Work done %d, old value: %d, pos %d, reg %04x\n", released, v, pos, reg);
-	sw_w32_mask(0x3ff << pos, released << pos, reg);
-	sw_w32(v, reg);
+	sw_w32(released << pos, reg);
```

Comment:
I don't think this is correct, this should be `sw_w32_mask(0x3ff << pos, released << pos, reg);` right? Otherwise, you are overwriting all other ring buffer counters with zero, effectively resetting them every time you update a single ring buffer ...

Reply:
To repeat my explanation "When you write X to a RING_CNTR it's value is decremented by X". A simple example:

```
Ring Counter                  0x12345678
sw_w32(0x00010000) gives      0x12335678 -> ok, one ring decremented
sw_w32_mask(0x00010000) gives 0x00030000 -> wrong, other rings reset

```